### PR TITLE
Adaptative (local) `abstol` to validate integration

### PR DESCRIPTION
**Pull request by lbenet**

### CODE REVIEW

**Comment by lbenet** [comment]
This PR aims to shrink the time step in order to validate the integration. It mimics what's currently done in [ReachabilityAnalysis.jl](https://github.com/JuliaReach/ReachabilityAnalysis.jl).

cc @mforets

**Comment by mforets** [comment]
Nice! I'll try this asap..

**Comment by lbenet** [comment]
By some  reason, some tests (in macos) do not pass, but after a restart of the tests, they do...

File: src/validatedODEs.jl [hunk 1]
```diff
@@ -51,8 +51,9 @@ function remainder_taylorstep!(f!::Function, t::Taylor1{T},
         Δx = Δ
     end
 
-    # If it didn't work, throw an error
-    @format full
+    # If it didn't work and adaptive is false, throw an error
+    adaptive && return (false, Δx, t[0])
```
Comment (schillic) [inline src/validatedODEs.jl:55]: The comment does not match.

Comment (mforets) [reply src/validatedODEs.jl:55]: @schillic if the program is outside the loop it means that "it didn't work", that the program couldn't prove the contraction property. then there are two options: either give up (if `adaptive` is false), or try again with another tolerance (`adaptive` is true). the first element in the tuple `(false, ...)` is a flag to say whether picard iteration succeeded or not.

Comment (schillic) [reply src/validatedODEs.jl:55]: Oh sorry, I see now: the error comes below :sweat_smile:

Comment (lbenet) [reply src/validatedODEs.jl:57]: Related to this comment: Shall we throw an error (and break the integration) or rather a warning, and (in the main function) stop continuing the integration due to this type of failures?

Comment (lbenet) [reply src/validatedODEs.jl:57]: (The same would apply in other parts of the code.)

**Comment by lbenet** [comment]
@mforets @UzielLinares The last two commits implement ideas we discussed (offline).

**Comment by lbenet** [comment]
Tests pass. Merging!

**Pull request merged**